### ISSUE TITLE
More detailed information on network errors

### ISSUE THREAD

**Issue by sdsantos** [blockstack/blockstack-android#174]
Right now, any kind of errors returned by an unsuccessful network call are returned generally as:

```
ResultError(ErrorCode.UnknownError, "Error when loading from Gaia hub, status:" + response.code())
```

That makes it harder to tell particular scenarios apart like a `404` or a `429`. To make matters a bit more confusing, the `ResultError` comes with a code parameter, but it's an error code, not a HTTP response code.

I suggest a Network sub-type of `ResultError` that passes along that information to the app. Right now we need to fallback on parsing the error message string created by the code above.

**Comment by friedger** [comment]
The `NetworkResultError` ideally has the network error code as code (in addition to the extra status code)

**Comment by friedger** [comment]
fixed by #175
